plugins/action/configure.py

```diff
@@ -0,0 +1,197 @@
+    return argspec
+
+
+class ActionModule(ActionBase):
+    def __init__(self, *args, **kwargs):
+        super(ActionModule, self).__init__(*args, **kwargs)
+        self._result = {}
+
+    def _fail_json(self, msg):
+        """ Replace the AnsibleModule fai_json here
+        :param msg: The message for the failure
+        :type msg: str
+        """
+        msg = msg.replace("(basic.py)", self._task.action)
+        raise AnsibleActionFail(msg)
+
+    def _check_argspec(self):
+        """ Load the doc and convert
+        Add the root conditionals to what was returned from the conversion
+        and instantiate an AnsibleModule to validate
+        """
+        argspec = generate_argspec()
+        basic._ANSIBLE_ARGS = to_bytes(
+            json.dumps({"ANSIBLE_MODULE_ARGS": self._task.args})
+        )
+        basic.AnsibleModule.fail_json = self._fail_json
+        basic.AnsibleModule(**argspec)
+
+    def _extended_check_argspec(self):
+        """ Check additional requirements for the argspec
+        that cannot be covered using stnd techniques
+        """
+        errors = []
+        config = self._task.args.get("config") or None
+        try:
+            # validate json
+            json.loads(config)
+        except Exception as exc:
+            msg = "Failed to load json configuration: %s" % (
+                to_text(exc, errors="surrogate_or_strict")
+            )
+            errors.append(msg)
+
+        yang_file = self._task.args.get("file") or None
+        if yang_file:
+            yang_file = os.path.realpath(os.path.expanduser(yang_file))
+            if not os.path.isfile(yang_file):
+                # Maybe we are passing a glob?
+                _yang_files = glob.glob(yang_file)
+                if not _yang_files:
+                    msg = "%s invalid yang_file path" % yang_file
+                    errors.append(msg)
+
+        if "search_path" in self._task.args:
+            search_path = self._task.args["search_path"]
+            for path in search_path.split(":"):
+                path = os.path.realpath(os.path.expanduser(path))
+                if path != "" and not os.path.isdir(path):
+                    msg = "%s is invalid search_path directory" % path
+                    errors.append(msg)
+        if errors:
+            self._result["failed"] = True
+            self._result["msg"] = " ".join(errors)
+
+    def run(self, tmp=None, task_vars=None):
+        """
+
+        :param terms:
+        :param variables:
+        :param kwargs:
+        :return:
+        """
+        self._check_argspec()
+        self._extended_check_argspec()
+        if self._result.get("failed"):
+            return self._result
+        result = super(ActionModule, self).run(tmp, task_vars)
+
+        json_config = self._task.args.get("config") or None
+        yang_file = self._task.args.get("file") or None
+        search_path = self._task.args.get("search_path") or None
+
+        if json_config:
```

Comment:
Move the file check login to translator module_utils `json_to_xml` method. Refer https://github.com/ansible-collections/community.yang/blob/main/plugins/module_utils/translator.py#L229-L249